src/client/BuilderClient.ts

```diff
@@ -30,6 +30,20 @@ export class BuilderClient {
     }
   }
 
+  /**
+   * Returns AuthIdentity either from the constant variable or the method given in the constructor.
+   */
+  private getIdentity(): AuthIdentity {
+    return this.identity instanceof Function ? this.identity() : this.identity
+  }
+
+  /**
+   * Returns an address either from the constant variable or the method given in the constructor.
+   */
+  private getAddress(): string {
+    return this.address instanceof Function ? this.address() : this.address
```

Comment:
I was thinking, how hard/worth it will be to have these a bit more explicit. By that I mean, we're having to deal with a variable that's either a function or a value, and the name is confusing as it's referring to both but `this.address()` looks weird.

SO, for your consideration. What if moving the logic to the constructor:

```
this._getAddress = () => { constructorGetAddress() || this.address }
```

(not an actual implementation). By that I mean just storing a function and letting it decide what to return instead of the private value


### this is just a suggestion, the current implementation works and it's fine by me